_posts/2024-02-07-#28950.md

```diff
@@ -8,16 +8,56 @@ authors: [instagibbs]
+
+- This PR adds optional parameters `maxfeerate` and `maxburnamount` to the [`submitpackage`](https://github.com/bitcoin-core-review-club/bitcoin/blob/bf6fbea9794db5e8d0adce8ff6182574808cea54/src/rpc/mempool.cpp#L816) RPC, enabling similar sanity checks on package transactions.
+
+- After [decoding each package transaction in `submitpackage`](https://github.com/bitcoin/bitcoin/blob/e69796c79c0aa202087a13ba62d9fbcc1c8754d4/src/rpc/mempool.cpp#L814), the transaction outputs are subjected to a `maxburnamount` sanity check.
+
+- The PR introduces the `max_sane_feerate` parameter to the [`ProcessNewPackage`](https://github.com/bitcoin-core-review-club/bitcoin/blob/bf6fbea9794db5e8d0adce8ff6182574808cea54/src/validation.h#L282) function. `submitpackage` now passes the value of `maxfeerate` as the `max_sane_feerate` when calling `ProcessNewPackage`.
+
+- `ProcessNewPackage` is updated to forward the `max_sane_feerate` to [`AcceptPackage`](https://github.com/bitcoin-core-review-club/bitcoin/blob/bf6fbea9794db5e8d0adce8ff6182574808cea54/src/validation.cpp#L1450), which performs the sanity check during [`Sub package`](https://github.com/bitcoin-core-review-club/bitcoin/blob/bf6fbea9794db5e8d0adce8ff6182574808cea54/src/validation.cpp#L1425) evaluation.
+    - Transactions with individual modified feerates exceeding `max_sane_feerate` will not accepted into the mempool, along with its descendants, as they will be missing input during their evaluation.
+
 ## Questions
+
 1. Did you review the PR? [Concept ACK, approach ACK, tested ACK, or NACK](https://github.com/bitcoin/bitcoin/blob/master/CONTRIBUTING.md#peer-review)? What was your review approach?
--->
+
+2. Why is it important to perform these sanity checks on submitted packages?
+
+3. Are there other important sanity checks apart from `maxburnamount` and `maxfeerate` that should be performed on packages before they are accepted to the mempool?
+
+4. The options `maxburnamount` and `maxfeerate` can prevent a transaction from entering the mempool and being relayed. Can we consider these options as policy rules? Explain why?
+
+5. The [bf6fbea97](https://github.com/bitcoin-core-review-club/bitcoin/commit/bf6fbea9794db5e8d0adce8ff6182574808cea54) commit message states: "This allows subpackage processing and is compatible with future package RBF work." What makes this compatible with future RBF work?
+
+6. Why do we validate `maxfeerate` against the modified feerate instead of the base fee rate? 
+
+7. We validate `maxfeerate` against the modified feerate of individual package transactions, not package feerate, this can sometimes be incentive incompatible.
+    - How does this happen?
+    - Why is it not a current concern? 
+    - Under what conditions might it become problematic?
+
+8. Why is `maxfeerate` not validated before package processing? 
```

Comment:
Maybe: "Why can't `maxfeerate` be checked immediately after decoding like `maxburnamount` is?"